python/ray/ml/predictors/integrations/torch/torch_predictor.py

```diff
@@ -0,0 +1,67 @@
+from typing import Optional, Union, List
+
+import torch
+
+from ray.ml.predictor import Predictor, DataBatchType
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+
+
+class TorchPredictor(Predictor):
+    """A predictor for PyTorch models.
+
+    Args:
+        model: The torch module to use for predictions.
+        preprocessor: The preprocessor used to transform data batches prior
+            to prediction.
+
+    """
+
+    def __init__(self, model: torch.nn.Module, preprocessor: Preprocessor):
+        raise NotImplementedError
+
+    @classmethod
+    def from_checkpoint(
+        cls, checkpoint: Checkpoint, model_definition: Optional[torch.nn.Module] = None
+    ) -> "TorchPredictor":
+        """Instantiate the predictor from a Checkpoint.
+
+        The checkpoint is expected to be a result of ``TorchTrainer``.
+
+        Args:
+            checkpoint: The checkpoint to load the model and
+                preprocessor from. It is expected to be from the result of a
+                ``TorchTrainer`` run.
+            model_definition: If checkpoints
+                contains the model state dict, and not the model itself,
+                then the state dict will be loaded to the model_definition.
+
+        """
+        raise NotImplementedError
+
+    def predict(
+        self,
+        data: DataBatchType,
+        column_names: Optional[List[str]] = None,
+        feature_columns: Optional[Union[List[str], List[List[str]]]] = None,
+        dtype: Optional[Union[torch.dtype, List[torch.dtype]]] = None,
+    ) -> DataBatchType:
+        """Run inference on data batch.
+
+        Args:
+            data: A batch of input data.
+            columns_names: If provided, the column names to set for the data
```

Comment:
What does that mean, to "set" them? Also, how does this relate to the next arg?

Reply:
+1

Reply:
`column_names` has been removed.